Bad Request (send2trash failed: [Errno 13] Permission denied: b'/.Trash-1000' when running the tests locally

This somehow relates to https://github.com/jupyter-server/jupyter_server/issues/1338

I have an up-to-date environment on a Ubuntu machine, generated with
```
cd jupytext
micromamba env create --file environment.yml
pip install -e .[dev]
```

The full stack trace is
```
______________________________________________ test_load_save_rename[ipynb_py/notebook_with_complex_metadata.ipynb-sync] ______________________________________________
[gw1] linux -- Python 3.13.2 /home/marc/micromamba/envs/jupytext-dev/bin/python3.13

volume_root = b'/'

    def find_ext_volume_fallback_trash(volume_root):
        # from [2] Trash directories (1) create a .Trash-$uid dir.
        trash_dir = op.join(volume_root, TOPDIR_FALLBACK)
        # Try to make the directory, if we lack permission, raise TrashPermissionError
        try:
>           check_create(trash_dir)

../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/send2trash/plat_other.py:158: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

dir = b'/.Trash-1000'

    def check_create(dir):
        # use 0700 for paths [3]
        if not op.exists(dir):
>           os.makedirs(dir, 0o700)
E           PermissionError: [Errno 13] Permission denied: b'/.Trash-1000'

../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/send2trash/plat_other.py:96: PermissionError

During handling of the above exception, another exception occurred:

self = <jupytext.sync_contentsmanager.build_sync_jupytext_contents_manager_class.<locals>.SyncJupytextContentsManager object at 0x7699091fe530>, path = 'new.Rmd'

    def delete_file(self, path):
        """Delete file at path."""
        path = path.strip("/")
        os_path = self._get_os_path(path)
        rm = os.unlink
    
        if not self.allow_hidden and is_hidden(os_path, self.root_dir):
            raise web.HTTPError(400, f"Cannot delete file or directory {os_path!r}")
    
        four_o_four = "file or directory does not exist: %r" % path
        if not self.exists(path):
            raise web.HTTPError(404, four_o_four)
    
        def is_non_empty_dir(os_path):
            if os.path.isdir(os_path):
                # A directory containing only leftover checkpoints is
                # considered empty.
                cp_dir = getattr(self.checkpoints, "checkpoint_dir", None)
                if set(os.listdir(os_path)) - {cp_dir}:
                    return True
    
            return False
    
        if self.delete_to_trash:
            if not self.always_delete_dir and sys.platform == "win32" and is_non_empty_dir(os_path):
                # send2trash can really delete files on Windows, so disallow
                # deleting non-empty files. See Github issue 3631.
                raise web.HTTPError(400, "Directory %s not empty" % os_path)
            # send2trash now supports deleting directories. see #1290
            if not self.is_writable(path):
                raise web.HTTPError(403, "Permission denied: %s" % path) from None
            self.log.debug("Sending %s to trash", os_path)
            try:
>               send2trash(os_path)

../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/jupyter_server/services/contents/filemanager.py:563: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/send2trash/plat_other.py:210: in send2trash
    dest_trash = find_ext_volume_trash(topdir)
../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/send2trash/plat_other.py:169: in find_ext_volume_trash
    trash_dir = find_ext_volume_fallback_trash(volume_root)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

volume_root = b'/'

    def find_ext_volume_fallback_trash(volume_root):
        # from [2] Trash directories (1) create a .Trash-$uid dir.
        trash_dir = op.join(volume_root, TOPDIR_FALLBACK)
        # Try to make the directory, if we lack permission, raise TrashPermissionError
        try:
            check_create(trash_dir)
        except OSError as e:
            if e.errno == errno.EACCES:
>               raise TrashPermissionError(e.filename)
E               send2trash.exceptions.TrashPermissionError: [Errno 13] Permission denied: b'/.Trash-1000'

../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/send2trash/plat_other.py:161: TrashPermissionError

The above exception was the direct cause of the following exception:

ipynb_py_R_jl_file = '/home/marc/GitHub/jupytext/tests/data/notebooks/inputs/ipynb_py/notebook_with_complex_metadata.ipynb'
cm = <jupytext.sync_contentsmanager.build_sync_jupytext_contents_manager_class.<locals>.SyncJupytextContentsManager object at 0x7699091fe530>
tmpdir = local('/tmp/pytest-of-marc/pytest-42/popen-gw1/test_load_save_rename_ipynb_py12')

    async def test_load_save_rename(ipynb_py_R_jl_file, cm, tmpdir):
        tmp_ipynb = "notebook.ipynb"
        tmp_rmd = "notebook.Rmd"
    
        cm.formats = "ipynb,Rmd"
        cm.root_dir = str(tmpdir)
    
        # open ipynb, save Rmd, reopen
        nb = jupytext.read(ipynb_py_R_jl_file)
        await ensure_async(cm.save(model=notebook_model(nb), path=tmp_rmd))
        nb_rmd = await ensure_async(cm.get(tmp_rmd))
        compare_notebooks(nb_rmd["content"], nb, "Rmd")
    
        # save ipynb
        await ensure_async(cm.save(model=notebook_model(nb), path=tmp_ipynb))
    
        # rename_file ipynb
        await ensure_async(cm.rename_file(tmp_ipynb, "new.ipynb"))
        assert not os.path.isfile(str(tmpdir.join(tmp_ipynb)))
        assert not os.path.isfile(str(tmpdir.join(tmp_rmd)))
    
        assert os.path.isfile(str(tmpdir.join("new.ipynb")))
        assert os.path.isfile(str(tmpdir.join("new.Rmd")))
    
        # delete one file, test that we can still read and rename_file it
>       await ensure_async(cm.delete("new.Rmd"))

tests/integration/contents_manager/test_contentsmanager.py:129: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/jupyter_server/services/contents/manager.py:489: in delete
    self.delete_file(path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jupytext.sync_contentsmanager.build_sync_jupytext_contents_manager_class.<locals>.SyncJupytextContentsManager object at 0x7699091fe530>, path = 'new.Rmd'

    def delete_file(self, path):
        """Delete file at path."""
        path = path.strip("/")
        os_path = self._get_os_path(path)
        rm = os.unlink
    
        if not self.allow_hidden and is_hidden(os_path, self.root_dir):
            raise web.HTTPError(400, f"Cannot delete file or directory {os_path!r}")
    
        four_o_four = "file or directory does not exist: %r" % path
        if not self.exists(path):
            raise web.HTTPError(404, four_o_four)
    
        def is_non_empty_dir(os_path):
            if os.path.isdir(os_path):
                # A directory containing only leftover checkpoints is
                # considered empty.
                cp_dir = getattr(self.checkpoints, "checkpoint_dir", None)
                if set(os.listdir(os_path)) - {cp_dir}:
                    return True
    
            return False
    
        if self.delete_to_trash:
            if not self.always_delete_dir and sys.platform == "win32" and is_non_empty_dir(os_path):
                # send2trash can really delete files on Windows, so disallow
                # deleting non-empty files. See Github issue 3631.
                raise web.HTTPError(400, "Directory %s not empty" % os_path)
            # send2trash now supports deleting directories. see #1290
            if not self.is_writable(path):
                raise web.HTTPError(403, "Permission denied: %s" % path) from None
            self.log.debug("Sending %s to trash", os_path)
            try:
                send2trash(os_path)
            except OSError as e:
>               raise web.HTTPError(400, "send2trash failed: %s" % e) from e
E               tornado.web.HTTPError: HTTP 400: Bad Request (send2trash failed: [Errno 13] Permission denied: b'/.Trash-1000')

../../micromamba/envs/jupytext-dev/lib/python3.13/site-packages/jupyter_server/services/contents/filemanager.py:565: HTTPError
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ Jupytext ChangeLog
 
 **Changed**
 - Jupytext's default contents manager is now derived from the asynchronous AsyncLargeFileManager. Thanks to [Darshan Poudel](https://github.com/Darshan808) for making this finally happen ([#1328](https://github.com/mwouts/jupytext/pull/1328))!
+- The [percent format](https://jupytext.readthedocs.io/en/latest/formats-scripts.html#the-percent-format) is now the default at the command line. If you run `jupytext --to py notebook.ipynb` you now get a `py:percent` script (use `--to py:light` for the light format) [#1201](https://github.com/mwouts/jupytext/pull/1201)
 - The `rst2md` conversion now works with `sphinx-gallery>=0.8`. Thanks to [Thomas J. Fan](https://github.com/thomasjpfan) for fixing this! ([#1334](https://github.com/mwouts/jupytext/pull/1334))
 - We have updated the JupyterLab extension dependencies ([#1300](https://github.com/mwouts/jupytext/pull/1300)). Thanks to [Mahendra Paipuri](https://github.com/mahendrapaipuri) for this PR!
 

--- a/tests/integration/contents_manager/test_contentsmanager.py
+++ b/tests/integration/contents_manager/test_contentsmanager.py
@@ -107,6 +107,7 @@ async def test_load_save_rename(ipynb_py_R_jl_file, cm, tmpdir):
 
     cm.formats = "ipynb,Rmd"
     cm.root_dir = str(tmpdir)
+    cm.delete_to_trash = False
 
     # open ipynb, save Rmd, reopen
     nb = jupytext.read(ipynb_py_R_jl_file)

